inc/azure_c_util/interlocked.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef INTERLOCKED
```

Comment:
>INTERLOCKED [](start = 8, length = 11)

INTERLOCKED_H - we are using FILENAME_H as a template for naming the token that prohibits double inclusion of the same header file.